mayastor/tests/nexus_multipath.rs

```diff
@@ -36,13 +36,17 @@ fn nvme_connect(target_addr: &str, nqn: &str) {
         .args(&["-n", nqn])
         .status()
         .unwrap();
-    assert!(
-        status.success(),
-        "failed to connect to {}, nqn {}: {}",
-        target_addr,
-        nqn,
-        status
-    );
+    if must_succeed {
+        assert!(
+            status.success(),
+            "failed to connect to {}, nqn {}: {}",
+            target_addr,
+            nqn,
+            status,
+        );
+    }
+
+    status
```

Comment:
Perhaps should still log a message to the effect that we failed to connect in the case that must_succeed is false.